### PR TITLE
[Paper] add comparison tables and benchmark

### PR DESCRIPTION
**Pull request by Santymax98**
Hi Oskar, I don't know what you think about this, I used "BenchmarkTools.jl" to get the results. Can it be helpful for correction?

### CODE REVIEW

**Comment by lrnv** [comment]
First table looks very good. The second one is surprising to me, this is the first time I ran BenchmarkTools on these things, and i NEVER optimized the code, this is wild that we beat DataGenCopulaBased like that. What about BivariateCopulas why dont you have a result ?

**Comment by Santymax98** [comment]
With "bivariatecopula.jl" it is only possible for the bivariate case and on the other hand I was not able to execute the code correctly. I was trying for quite some time using the documentation but I didn't succeed. Maybe it's a problem with my machine. On the other hand, do you think we should cite "BenchmarkTools" and put the characteristics of the machine I used to obtain those results?

**Comment by lrnv** [comment]
That is good ! Yes you can cite benchmarktools like i cited other packages: with a link to their repository +, if they have some bibtex citation in their readme and/or docs, these bibtex entries. Look at what I did for BivariateCopulas to see how it works. 

Nice numbers. Maybe you could state somthing before like "Since our primary target is  maintainability and readability of the implementation, we did not consider the efficiency and the perfromance of the code yet. However, a (limited in scope) benchmark on Copulas.jl's implementation of Clayton's pdf show that bla bla bla".

File: joss/paper.md [hunk 1]
```diff
@@ -40,6 +40,27 @@ The Julia package `Copulas.jl` brings most standard copula-related features into
...
+# Comparison of other packages with copulas.jl
+According to the package documentation, it is possible to summarize some of the most important functionalities of each package and those most needed by the community.
+## Functionality
+The following table shows some characteristics that differentiate each package.
+| Characteristic                                | Copulas.jl         | DatagenCopulaBased.jl         | BivariateCopulas.jl          |
+|-----------------------------------------------|--------------------|--------------------|--------------------|
+| Every Archimedean Copula sampling     | Yes                 | No                 | No                 |
```
Comment (lrnv) [inline joss/paper.md:49]: You could add here a line "Classic Bivariate copulas sampling" with Yes/Yes/Yes. 

Also a line "Obscure bivariate copulas sampling" with "Yes/No/No ?" ^^

Maybe the first column could be refactored to : 
Sampling
    - Classic Bivariate Copulas  Yes Yes Yes 
    - Obscure Bivariate Copulas Yes No No
    - Archimedean copulas All / Classic Only / Classic Only
    - Multivariate Copulas Yes Yes No (You exchanged the two i think)
    - Archimedean Chains (this is the real name of "nested" : No No Yes
Fitting   No No Yes
Plotting No No Yes
Dependence metrics Partial / ?  / ? (check)

**Comment by lrnv** [comment]
This will fix #116 when merged.

**Comment by lrnv** [comment]
It would also be nice to add this benchmark to the documentation (with the code to run it) since it is pretty :)

File: joss/paper.md [hunk 2]
```diff
@@ -46,21 +46,25 @@ According to the package documentation, it is possible to summarize some of the
...
+| Multivariate Copula                           | Yes                | Yes                | No                 |
+| Archimedean Chains                            | No                 | Yes                | No                 |
+| Fitting Copula                                | No                 | No                 | Yes                |
+| Plotting Copula                               | No                 | No                 | Yes                |
 ##  Efficiency
-To perform an efficiency test we use the "BenchmarkTools" package with the objective of comparing the execution time and the amount of memory necessary to generate copula samples with each package. We generate 10^6 samples for Clayton copula of dimensions 2, 5, 10 with parameter 0.8
+To perform an efficiency test we use the [`BenchmarkTools.jl`](https://github.com/JuliaCI/BenchmarkTools.jl) [@BenchmarkTools] package with the objective of comparing the execution time and the amount of memory necessary to generate copula samples with each package. We generate 10^6 samples for Clayton copula of dimensions 2, 5, 10 with parameter 0.8
```
Comment (lrnv) [inline joss/paper.md:58]: For the reference to work you need to add it to the .bib file, taking it from their CITATION file there : https://github.com/JuliaCI/BenchmarkTools.jl/blob/master/CITATION.bib

@ARTICLE{BenchmarkTools,
  author =	 {{Chen}, Jiahao and {Revels}, Jarrett},
  title =	 "{Robust benchmarking in noisy environments}",
  journal =	 {arXiv e-prints},
  keywords =	 {Computer Science - Performance, 68N30, B.8.1, D.2.5},
  year =	 2016,
  month =	 "Aug",
  eid =		 {arXiv:1608.04295},
  archivePrefix ={arXiv},
  eprint =	 {1608.04295},
  primaryClass = {cs.PF},
  adsurl =	 {https://ui.adsabs.harvard.edu/abs/2016arXiv160804295C},
  adsnote =	 {Provided by the SAO/NASA Astrophysics Data System}
}

**Comment by Santymax98** [comment]
Hello, I added the code, however I only added the one I used for copulas.jl but it works with any package. Additionally, it must be taken into account that the methodology for obtaining samples is different for each package.

**Comment by lrnv** [comment]
Thanks, this is Ok for now I will finsh this PR and merge it myself on monday. very good job. 

Can you now focus on #98 from now on ? I am sure this is not very hard to solve but this needs to work correctly.

**Comment by Santymax98** [comment]
Do you think I can add a thank you to the foundation that finances my
studies?

**Comment by Santymax98** [comment]
Thanks, I will.

**Pull request merged**